object/token.go

```diff
@@ -608,3 +611,59 @@ func GetTokenByUser(application *Application, user *User, scope string, host str
 	AddToken(token)
 	return token, nil
 }
+
+// Wechat Mini Program flow
+func GetWechatMiniProgramToken(application *Application, code string, providerName string, host string) (*Token, error) {
+	provider := GetProvider(util.GetId(providerName))
+	mpIdp := idp.NewWeChatMPIdProvider(provider.ClientId, provider.ClientSecret)
+	session, err := mpIdp.GetSeesionByCode(code)
+	if err != nil {
+		return nil, err
+	}
+	openId, unionId := session.Openid, session.Unionid
+	if openId == "" && unionId == "" {
+		return nil, errors.New("err: openid and unionid is empty")
+	}
+	user := getUserByOpenId(openId, unionId)
+	if user == nil {
+		//Add new user
+		user = &User{
+			Owner:             application.Organization,
+			Id:                util.GenerateId(),
+			Name:              fmt.Sprintf("wechat-%s", openId),
+			SignupApplication: application.Name,
+			WeChat:            openId,
+			Unionid:           unionId,
+			Type:              "normal-user",
+			CreatedTime:       util.GetCurrentTime(),
+			IsAdmin:           false,
+			IsGlobalAdmin:     false,
+			IsForbidden:       false,
+			IsDeleted:         false,
+		}
+		AddUser(user)
+	}
+
+	accessToken, refreshToken, err := generateJwtToken(application, user, "", "", host)
+	if err != nil {
+		return nil, err
+	}
+
+	token := &Token{
```

Comment:
I think the JWT token can add a ClientID field, which storage  the openId or unionId